js/sparkswap.js

```diff
@@ -271,7 +271,46 @@ module.exports = class sparkswap extends Exchange {
     }
 
     async fetchOrder (id, symbol = undefined, params = {}) {
-        return this.privateGetV1OrderId ({ id });
+        const order = await this.privateGetV1OrdersId ({ id });
+        let status = 'open';
+        if (order.status === 'CANCELLED') {
+            status = 'cancelled';
+        }
+        if (order.status === 'COMPLETED') {
+            status = 'closed';
+        }
+        let type = 'limit';
+        if (order.is_market_order) {
+            type = 'market';
+        }
+        let side = 'buy';
+        if (order.side === 'ASK') {
+            side = 'sell';
```

Comment:
Same thing here. I feel like if we have the information why not actually look at it? If there is something wrong with the order and the order.side is not ask or bid, we will be setting the type to buy when there is something wrong with the data.